subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/client/kubernetes/ReservationDeploymentFactory.kt

```diff
@@ -125,13 +125,23 @@ class ReservationDeploymentFactory(
                     }
 
                     resources {
-                        limits = mapOf(
-                            "cpu" to Quantity(emulator.cpuCoresLimit),
-                            "memory" to Quantity(emulator.memoryLimit)
-                        )
-                        requests = mapOf(
-                            "cpu" to Quantity(emulator.cpuCoresRequest)
-                        )
+                        limits = mutableMapOf<String, Quantity>().apply {
+                            if (!emulator.cpuCoresLimit.isNullOrBlank()) {
+                                plusAssign("cpu" to Quantity(emulator.cpuCoresLimit))
+                            }
+                            if (!emulator.memoryLimit.isNullOrBlank()) {
+                                plusAssign("memory" to Quantity(emulator.memoryLimit))
+                            }
+                        }
+
+                        requests = mutableMapOf<String, Quantity>().apply {
+                            if (!emulator.cpuCoresRequest.isNullOrBlank()) {
+                                plusAssign("cpu" to Quantity(emulator.cpuCoresRequest))
+                            }
+                            if (!emulator.memoryRequest.isNullOrBlank()) {
+                                plusAssign("memory" to Quantity(emulator.memoryRequest))
+                            }
+                        }
```

Comment:
Do we have an implicit contract for requests and limits?
E.g. are pods without limits and requests make sense and valid?

Reply:
If I see an original error from k8s, it will be enough.

Reply:
Yes and no; We could create it for some test purposes, but in real load it should be set